recurring_contract/models/contract_group.py

```diff
@@ -247,14 +260,20 @@ def _clean_generate_invoices(self):
         contract group.
         """
         res = self.env['account.invoice']
+        since_date = date.today()
         for group in self:
-            since_date = date.today()
-            if group.last_paid_invoice_date:
-                last_paid_invoice_date = group.last_paid_invoice_date
-                since_date = max(since_date, last_paid_invoice_date)
```

Comment:
the if wasn't used in the last version of the code. (neither since date nor last paid invoice are used) 

rewind_next_invoice_date handle the logic to regenerate invoices from today.